### PR TITLE
reprimand: New package

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Comment by eschnett** [comment]
ping

**Comment by eschnett** [comment]
ping ping

File: var/spack/repos/builtin/packages/reprimand/package.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    depends_on('boost')
+    depends_on('gsl')
+    depends_on('hdf5')
+    depends_on('python', when='+python')
+    depends_on('py-matplotlib', when='+python')
+    depends_on('py-pybind11 @2.6.0:', when='+python')
+
+    # Should we add this?
+    # extends('python')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/reprimand/package.py]: You can say `extends('python', when="+python")`, but did you have other issues with this?

File: var/spack/repos/builtin/packages/reprimand/package.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    def setup_build_environment(self, env):
+        env.set('CXXFLAGS', self.compiler.cxx11_flag)
+        env.set('BOOST_ROOT', self.spec['boost'].prefix)
+        env.prepend_path('PKG_CONFIG_PATH', self.spec['gsl'].prefix)
+        env.prepend_path('PKG_CONFIG_PATH', self.spec['hdf5'].prefix)
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/reprimand/package.py]: (no change required) By default we add `<prefix>/{lib, lib64, share}/pkgconfig` but I'm guessing that wasn't sufficient here? Perhaps we should be dynamically looking for whatever directories contain `.pc` files and add those.

**Review by scheibelp — COMMENTED** [review]
You had a question about `extends(python)` which I added a comment on. Since this is a library, if you want the Python bindings added to `PYTHONPATH` etc. then I think you'll want to do that before this is merged. Generally this looks good.

File: var/spack/repos/builtin/packages/reprimand/package.py (hunk 1 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/reprimand/package.py]: Done. Thanks!

File: var/spack/repos/builtin/packages/reprimand/package.py (hunk 2 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/reprimand/package.py]: These were actually not needed.

**Review by scheibelp — APPROVED** [review]

**Pull request merged**